NCSDK-28262: Pass encryption info to the SUIT platform

tomchy opened the pull request.

tomchy committed:
platform: Pass encryption info to the platform

SUIT procssor is responsible for decoding encryption info.
The SUIT platform is responsible for performing the decryption.
Pass the encryption info through the dedicated structure to the
platform, so it can perform the decryption.

Ref: NCSDK-28262

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

NordicBuilder commented:
Posix Twister tests results

<a href='https://jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-dfu/job/master/90/artifact/twister_posix/coverage/index.html'>Detailed report</a>:
| Type | Coverage |
| --- | --- |
| lines | 90.7% (1499 of 1653 lines) |
| functions | 98.8% (81 of 82 functions) |
| branches | no data found |


*Note: This message is automatically posted and updated by the CI (latest/test-sdk-dfu/master/90)*

tomchy committed:
tests: Add test for encryotion info parameter

Add a basic test that verifies that encryption info can be parsed and
set through override-parameters directive.

Ref: NCSDK-28258

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>
